src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderChapter.jsx

```diff
@@ -0,0 +1,378 @@
+ * @property {function(FormConfigPage): FormConfigPage} [itemFirstPage] First page for the item
+ * @property {function(FormConfigPage): FormConfigPage} [itemMiddlePage] A page between the first and last page for the item
+ * @property {function(FormConfigPage): FormConfigPage} [itemLastPage] Last page for the item that will navigate back to the summary page
+ * @property {function(FormConfigPage): FormConfigPage} [itemSinglePage] Use this if there's only one page
+ */
+
+/**
+ * @typedef {Object} ArrayBuilderOptions
+ * @property {string} nounSingular e.g. `employer`
+ * @property {string} nounPlural e.g. `employers`
+ * @property {string} arrayPath e.g. `employers`
+ * @property {string} nextChapterPath e.g. `/next-chapter`
+ * @property {string} [reviewPath] e.g. `/review-and-submit`
+ * @property {(item) => boolean} [isIncompleteItem] Will display error on the cards if item is incomplete. e.g. `item => !item?.name`
+ */
+
+/**
+ * @typedef {FormConfigChapter & {
+ *   options: ArrayBuilderOptions
+ * }} ArrayBuilderConfig
+ */
+
+function getBasePath(path) {
+  const basePath = path.split('/:')[0];
+  return basePath.startsWith('/') ? basePath : `/${basePath}`;
+}
+
+function throwErrorPage(pageType, option) {
+  throw new Error(
+    `arrayBuilderChapter \`pages.${pageType}()\` must include \`${option}\` property like this: ` +
+      `\`arrayBuilderChapter(pages => ({ examplePage: pages.${pageType}({ ${option}: ... }) }))\``,
+  );
+}
+
+function throwErrorChapter(option) {
+  throw new Error(
+    `arrayBuilderChapter must include \`options.${option}\` property`,
+  );
+}
+
+function throwErrorNoConfigFunction() {
+  throw new Error(
+    'arrayBuilderChapter must include a config function like this `arrayBuilderChapter(pages => { ... })`',
+  );
+}
+
+function throwErrorNoOptions() {
+  throw new Error(
+    'arrayBuilderChapter must include an `options` property like this: ' +
+      '`arrayBuilderChapter(pages => ({ options: { ... } }))`',
+  );
+}
+
+function throwMissingYesNoField() {
+  throw new Error(
+    "arrayBuilderChapter `pages.summaryPage()` must include a `uiSchema` that has a property with a `'ui:webComponentField': YesNoField`",
+  );
+}
+
+function throwIncorrectItemPath() {
+  throw new Error(
+    'arrayBuilderChapter item pages must include a `path` property that ends with `/:index`',
+  );
+}
+
+function verifyRequiredOptions(pageType, requiredOpts, objectToCheck) {
+  requiredOpts.forEach(option => {
+    if (objectToCheck[option] === undefined) {
+      if (pageType === 'chapter') {
+        throwErrorChapter(option);
+      } else {
+        throwErrorPage(pageType, option);
+      }
+    }
+  });
+}
+
+function determineYesNoField(uiSchema) {
+  let yesNoKey;
+  for (const key of Object.keys(uiSchema)) {
+    if (uiSchema[key]['ui:webComponentField'] === YesNoField) {
+      yesNoKey = key;
+    }
+  }
+  return yesNoKey;
+}
+
+/**
+ * arrayBuilderChapter
+ *
+ * Example:
```

Comment:
This comment is good to review see the big picture of this PR. 
This is the proposal for how the consumer will use this.